apps/src/applab/firebaseStorage.js

```diff
@@ -249,36 +254,51 @@ FirebaseStorage.deleteRecord = function (tableName, record, onComplete, onError)
  * @param {string} tableName Table to listen to.
  * @param {function (Object, RecordListener.EventType)} onRecord Callback to call when
  * a change occurs with the record object (described above) and event type.
+ * @param {function (string)} onWarning Callback to call with an warning to show to the user.
  * @param {function (string, number)} onError Callback to call with an error to show to the user and
  *   http status code.
  */
-FirebaseStorage.onRecordEvent = function (tableName, onRecord, onError) {
+FirebaseStorage.onRecordEvent = function (tableName, onRecord, onWarning, onError, includeAll) {
   if (typeof onError !== 'function') {
     throw new Error('onError is a required parameter to FirebaseStorage.onRecordEvent');
   }
   if (!tableName) {
     onError('Error listening for record events: missing required parameter "tableName"', 400);
     return;
   }
+  if (listenedTables.includes(tableName)) {
+    onWarning(`onRecordEvent was already called for table "${tableName}". To avoid 
+unexpected behavior in your program, you should only call onRecordEvent once 
+per table and use if/else statements to handle the different event types.`);
+  }
+  listenedTables.push(tableName);
+
+  getLastRecordId(tableName).then(lastId => {
+    const recordsRef = getRecordsRef(Applab.channelId, tableName);
+    // CONSIDER: Do we need to make sure a client doesn't hear about updates that it triggered?
```

Comment:
I should add that it seems reasonably easy to filter out messages from your own client. For example in the chat program, each chat message would have a sender name, and so you could ignore chat messages from your own client that way if you wanted.